plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php

```diff
@@ -238,13 +238,30 @@ public function prepare_items() {
 		 *
 		 * @param array $query_args Arguments to be passed to `wc_get_orders()`.
 		 */
-		$orders      = wc_get_orders( (array) apply_filters( 'woocommerce_order_list_table_prepare_items_query_args', $this->order_query_args ) );
+		$order_query_args = (array) apply_filters( 'woocommerce_order_list_table_prepare_items_query_args', $this->order_query_args );
+
+		// We must ensure the 'paginate' argument is set.
+		$order_query_args['paginate'] = true;
```

Comment:
This isn't directly related to the problem at hand, but I thought it was a sensible safeguard (since custom code hooked in to the above filter could remove it, but we depend on it being set to true).